### ISSUE TITLE
mpv is not compiled with vaapi support

### ISSUE THREAD

**Issue by tjikini** [NixOS/nixpkgs#26786]
## Issue description

mpv is not compiled with vaapi support

### Steps to reproduce

```
nix-env -Ai nixos.mpv
mpv -hwdec=vaapi /media/movies/movie.mkv
```
Output says:
`Requested hardware decoder not compiled.`

## Technical details

* System: NixOS: 17.09pre109180.0d4431cfe9
* Nix version: 1.11.10
* Nixpkgs version: 17.09pre109180.0d4431cfe9
* Sandboxing enabled: build-use-sandbox = true

**Comment by rnhmjoj** [comment]
Ok, this took me a while to figure. It is compiled with vaapi ([source](https://github.com/NixOS/nixpkgs/blob/c49a454210b92f45915d591cce88bd4b12eeb5b3/pkgs/applications/video/mpv/default.nix#L33)) but the vaapiSupport option (along with bs2bSupport, youtubeSupport, cacaSupport) is overridden in all-packages.nix where the package is called. @AndersonTorres added this and i'm not sure what's the motivation.

Removing the default arguments seems to fix this issue:
```diff
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index cbe2a9bfc5..9ec529d41a 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -15030,11 +15030,6 @@ with pkgs;
   mpv = callPackage ../applications/video/mpv rec {
     lua = lua5_1;
     lua5_sockets = lua5_1_sockets;
-    youtube-dl = pythonPackages.youtube-dl;
-    bs2bSupport = config.mpv.bs2bSupport or true;
-    youtubeSupport = config.mpv.youtubeSupport or true;
-    cacaSupport = config.mpv.cacaSupport or true;
-    vaapiSupport = config.mpv.vaapiSupport or false;
     libva = libva-full;
   };
```

cc @fuuzetsu @fpletz

**Comment by tjikini** [comment]
Thanks @rnhmjoj ! I did not expect such overriding in all-packages.nix :)

**Comment by Fuuzetsu** [comment]
the way to fix this issue is to set `mpv.vappiSupport` to true in your nix config, not by removing default arguments from all-packages.nix

**Comment by rnhmjoj** [comment]
Shouldn't packages be configured using `.override`? mpv is the only package that I have seen doing this way.

**Comment by Fuuzetsu** [comment]
many packages do this, just grep for `config.` in all-packages; I don't know if there's a policy in how to do this these days, not that I'm aware of at least.

**Comment by rnhmjoj** [comment]
I see. Anyway we should either remove these or add a `config.mpv.<option>` for all the others. It doesn't make much sense to have only a few options configurable this way.

**Comment by AndersonTorres** [comment]
My two cents on my first expression on NixOS :)

I have made the `config.mpv` options a long time ago, when most of packages were configured that way; in fact, I have just followed ffmpeg expression as an example. Today we don't use `config.ffmpeg` anymore.

As far as I know, there's no policy in special about it in Nixpkgs. I will "fix" it now.

**Comment by fpletz** [comment]
Closed by #26865.
